tests/unit/Diff/Internal/AliasGroupListPatcherTest.php

```diff
@@ -0,0 +1,229 @@
+<?php
+
+namespace Wikibase\DataModel\Services\Tests\Diff\Internal;
+
+use Diff\DiffOp\Diff\Diff;
+use Diff\DiffOp\DiffOpAdd;
+use Diff\DiffOp\DiffOpChange;
+use Diff\DiffOp\DiffOpRemove;
+use PHPUnit_Framework_TestCase;
+use Wikibase\DataModel\Services\Diff\Internal\AliasGroupListPatcher;
+use Wikibase\DataModel\Term\AliasGroup;
+use Wikibase\DataModel\Term\AliasGroupList;
+
+/**
+ * @covers Wikibase\DataModel\Services\Diff\Internal\AliasGroupListPatcher
+ *
+ * @licence GNU GPL v2+
+ * @author Bene* < benestar.wikimedia@gmail.com >
+ */
+class AliasGroupListPatcherTest extends PHPUnit_Framework_TestCase {
+
+	public function providePatchAliasGroupList() {
+		return array(
+			'add aliases' => array(
+				new AliasGroupList(),
+				new Diff( array(
+					'en' => new Diff( array( new DiffOpAdd( 'foo' ), new DiffOpAdd( 'bar' ) ) )
+				) ),
```

Comment:
I notice there is never a true/false here, so all `Diff` instances rely on the "is associative" auto-detection. I would either avoid that or add tests for all three possible values true/false/null.